corehq/sql_db/connections.py

```diff
@@ -191,26 +197,31 @@ def close_scoped_sessions(self):
         for helper in self._session_helpers.values():
             helper.Session.remove()
 
-    def dispose_engine(self, engine_id=DEFAULT_ENGINE_ID):
+    def dispose_engines(self, engine_id=DEFAULT_ENGINE_ID):
         """
         If found, closes the active sessions associated with an an engine and disposes it.
         Also removes it from the session manager.
         If not found, does nothing.
         """
-        self._dispose_engine(self.get_connection_string(engine_id))
+        self._dispose_engines_for_url(self.get_connection_string(engine_id))
 
-    def _dispose_engine(self, connection_string):
-        if connection_string in self._session_helpers:
-            helper = self._session_helpers.pop(connection_string)
-            helper.Session.remove()
-            helper.engine.dispose()
+    def _dispose_engines_for_url(self, connection_url):
+        for key in list(self._session_helpers):
+            (url, args) = key
+            if url == connection_url:
+                self.__dispose(key)
+
+    def __dispose(self, key):
```

Comment:
did you intentionally use two `_`? Just curious if we had a standard for that since I don't think I've seen it before